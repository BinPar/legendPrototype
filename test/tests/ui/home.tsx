import React from 'react';
import { mount } from 'enzyme';
import mockNextUseRouter from '../../mocks/mockNextUseRouter';
import Index from '../../../pages/index';

/**
 * Checks that the Home Page redirects to some sample
 */
test('Home Page', () => {

  /**
   * Stores the last route used by the router
   * updated when a push is received
   */
  let lastPath = '';


  /**
   * Mocks the next router
   */
  mockNextUseRouter({
    route: '/',
    pathname: '/',
    query: {},
    asPath: '/',
    push: (_, path): void => {
      lastPath = path;
    },
  });

  // Renders the page
  const page = mount(<Index />);

  // The home page have no text on it, just redirects to some page
  expect(page.text()).toMatch('');  

  // The router detect a pushed redirection to some sample URL
  expect(lastPath.indexOf('/sample/')).toBe(0);

});
