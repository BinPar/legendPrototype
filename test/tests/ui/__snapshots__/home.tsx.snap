// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page 1`] = `
<Component>
  <Head>
    <Component
      handleStateChange={null}
      inAmpMode={false}
      reduceComponentsToState={[Function]}
    />
  </Head>
</Component>
`;
